backend.dev.Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM node:alpine3.17 AS builder
+WORKDIR /app
+
+COPY packages/server/package.json ./
+COPY yarn.lock ./
+RUN yarn remove node-cron @joystream/types
+RUN yarn --immutable
+
+COPY tsconfig.json ./base.tsconfig.json
+COPY packages/server/tsconfig.json ./
+RUN sed -i 's/"extends":.*/"extends": ".\/base.tsconfig.json",/' tsconfig.json
+
+COPY packages/server/prisma ./
+COPY packages/server/codegen.ts ./
+COPY packages/server/src ./src
+RUN yarn build:local
+
+RUN rm -rf node_modules
+ENV NODE_ENV=production
+RUN yarn --prod --immutable
+RUN yarn prisma generate
```

Comment:
Why is this double `yarn` install necessary?

Reply:
Basically to reduce the final image size:
Some packages are huge due to their dependencies especially prisma however due to prisma it's really hard to bundle the code without the `node_modules` in the final image. So the first install create a `node_modules` with all the dev dependencies which also includes what's needed by the [`yarn build:local`](https://github.com/Joystream/pioneer/blob/backend/packages/server/package.json#L29C6-L29C11) script, then a much smaller `node_modules` is installed with only what is needed to run the code and the migrate db. Because the first install is done in a build stage it doesn't increase the image size.

I think the size reduction from this is pretty big the first version of the image was close to 1GB.